pkg/auth/dependency/interaction/adaptors/identity.go

```diff
@@ -135,6 +141,26 @@ func (a *IdentityAdaptor) New(userID string, typ authn.IdentityType, claims map[
 	panic("interaction_adaptors: unknown identity type " + typ)
 }
 
+func (a *IdentityAdaptor) NewForUpdate(identityID string, userID string, typ authn.IdentityType, claims map[string]interface{}) (*interaction.IdentityInfo, error) {
```

Comment:
This method could be changed to `WithClaims(ii *interaction.IdentityInfo, claims map[string]interface{}) (*interaction.IdentityInfo, error)`. The caller should first call `Get` and then call `WithClaims`.

Reply:
Updated in 9b0c9d6, see if it looks good